### PR TITLE
docs: add dynamic stats to routing recommendations

### PR DESCRIPTION
**Pull request by thepassle**
Preview deploy: https://deploy-preview-1147--open-wc-org.netlify.com/developing/routing.html

This adds a custom vue component to the routing recommendations so we can show some extra stats like github stars and downloadscount. It also links through to npmtrends.

We could also add a notification if a downloads count falls below a certain treshold, and we should discuss what this treshold should be.

image

If the request failed:

image

### CODE REVIEW

File: docs/.vuepress/components/NpmUsage.vue [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+      Loading...
+    </div>
+
+    <div v-if="rejected">
+      <p>Unable to get usage statistics for {{ routerName }}</p>
+    </div>
+
+    <div v-if="success">
+      <p>
+        <b>Usage stats:</b> {{ usageCount }} downloads (<a target="_blank" :href="'https://www.npmtrends.com/' + routerName">see more</a>)
```
Comment (LarsDenBakker) [inline docs/.vuepress/components/NpmUsage.vue]: Downloads weekly?

File: docs/.vuepress/components/NpmUsage.vue [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      </p>
+      <p>
+        <b>Github stars:</b> {{ stars }}
+      </p>
+      <p v-if="usageCount < 100">
```
Comment (LarsDenBakker) [inline docs/.vuepress/components/NpmUsage.vue]: Do you any rationale behind 100?

Comment (thepassle) [reply docs/.vuepress/components/NpmUsage.vue]: No, I just went for a random number, thats why I mentioned in the OP that we should discuss what we should base this treshold on 😛

**Comment by LarsDenBakker** [comment]
The UI is a bit jumpy as it loads the data, perhaps we can show the labels already while loading?

**Comment by thepassle** [comment]
With labels do you mean the names of the library?

**Comment by LarsDenBakker** [comment]
`Usage stats:` and `Github stars:`, the static content is shown way before the vue components are.

**Comment by thepassle** [comment]
> `Usage stats:` and `Github stars:`, the static content is shown way before the vue components are.

fixed

**Pull request merged**